logstash-core/lib/logstash/filter_delegator.rb

```diff
@@ -44,9 +44,9 @@ def config_name
     def multi_filter(events)
       @metric_events_in.increment(events.size)
 
-      start_time = java.lang.System.current_time_millis
+      start_time = java.lang.System.nano_time
       new_events = @filter.multi_filter(events)
-      @metric_events_time.increment(java.lang.System.current_time_millis - start_time)
+      @metric_events_time.increment((java.lang.System.nano_time - start_time) / 1_000_000)
```

Comment:
Is it possible for `nano_time - start_time` to be negative (say when NTP has caused clock to jump "back")? Can there be a check which calls increment only if the difference is positive?

Reply:
Java's nanotime does not use wallclock time (wallclock is affected by ntp adjustments). It is not expected to go backwards.

From the [System.nanoTime docs](https://docs.oracle.com/javase/8/docs/api/java/lang/System.html#nanoTime--):
> This method can only be used to measure elapsed time and is not related to any other notion of system or wall-clock time

Given this, it should be safe to use for duration measurements (where currentTimeMillis is not because it is based on wallclock time and impacted by ntp)

Reply:
@jordansissel That makes sense. Thanks for clarifying.